role-based-access-system/src/routes/user.route.js

```diff
@@ -0,0 +1,45 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+//get role for user
+route.get('/role/user/:userId', async (req, res) => {
+    try {
+        const userId = req.params.userId;
+        const getUserRole = await query(`SELECT r.name 
+        FROM roles r
+        INNER JOIN user u 
+        ON u.id=r.user_id
+        WHERE u.id=${userId}`)
+        if (getUserRole.affectedRows === 0) {
+            return res.send({ statusCode: 404, message: "Data not found" });
+        }
+        return res.send({ statusCode: 200, message: "Get Role of User", data: getUserRole })
+    }
+    catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+})
+
+//get permission for user
+route.get('/permission/user/:userId', async (req, res) => {
```

Comment:
Why permission route in user.routes.js ?

Reply:
made chnages in api url